docs/components/tree-list.md

```diff
@@ -0,0 +1,429 @@
+
+<div>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList class="mt-2" v-model="myList" />
+  <br>
+  <KButton @click="reset">Reset</KButton>
+</div>
+
+```html
+<template>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <KButton @click="reset">Reset</KButton>
+</template>
+
+<script lang="ts" setup>
+import { ref } from 'vue'
+
+const myList = ref([{
+  name: "Cats",
+  id: 'cats'
+  selected: true
+},
+{
+  name: "Dogs",
+  id: 'dogs',
+  children: [{
+    name: "Puppies",
+    id: 'puppies'
+  }]
+},
+{
+  name: "Bunnies",
+  id: 'bunnies'
+}])
+
+const reset = () => {
+  myList.value = [{
+    name: "Cats",
+    id: 'cats',
+    selected: true
+  },
+  {
+    name: "Dogs",
+    id: 'dogs',
+    children: [{
+      name: "Puppies",
+      id: 'puppies'
+    }]
+  },
+  {
+    name: "Bunnies",
+    id: 'bunnies'
+  }]
+}
+</script>
+```
+
+### items
+
+An array of items that make up the tree.
+
+Item properties:
+
+- `name` (required) - text displayed as the label for the item
+- `id` (required) - a unique `string` used to identify the item
+- `selected` - boolean to indicate whether the current item is selected or not
+- `icon` - `KIcon` name to be displayed to the left of the item `name` (defaults to `treeDoc`, specify `none` to not display any icon)
+- `children` - an array of items that will be styled as children of the current item
+
+::: danger
+You cannot use `v-model` with the `items` prop. You must use one or the other.
+:::
+
+<KTreeList :items="defaultItems2" />
+
+```html
+<template>
+  <KTreeList :items="items" />
+</template>
+
+<script lang="ts" setup>
+import { ref } from 'vue'
+
+const items = ref([{
+  name: "Cats",
+  id: 'cats'
+},
+{
+  name: "Dogs",
+  id: 'dogs',
+  children: [{
+    name: "Puppies",
+    id: 'puppies'
+  }]
+},
+{
+  name: "Bunnies",
+  id: 'bunnies'
+}])
+</script>
+```
+
+### disableDrag
+
+Boolean (defaults to `false`) to turn of drag-n-drop reordering of the list.
+
+<KTreeList disable-drag :items="disableItems" />
+
+```html
+<KTreeList disable-drag :items="items" />
+```
+
+### maxLevel
+
+Use this prop to customize the maximum supported depth of the tree. We default to a max level of `2` or support for parents, children, and grandchildren.
+
+<KTreeList :items="maxLevelItems" :max-level="1" />
+
+```html
+<KTreeList :items="items" :max-level="1" />
+```
+
+## Slots
+
+`KTreeList` allows you to customize individual tree items via the item slots. These slots return the current `item` data as a slot param.
+
+- `item-icon` - slot for content displayed to the left of the item name
+- `item-label` - slot for the main content of an item (default's to display the `name` of the item)
+
+<KTreeList :items="slotItems">
+  <template #item-icon="{ item }">
+    {{ item.id === 'cats' ? '😸' : item.id === 'bunnies' ? '🐰' : '🐶' }}
+  </template>
+  <template #item-label="{ item }">
+    <span class="color-purple-400">
+    {{ item.name }}
+    </span>
```

Comment:
suggestion: To show that you're slotting something here besides just a color change, what if you added a prefix to the name or something as well?
```suggestion
    <span class="color-purple-400">
    Animal: {{ item.name }}
    </span>
```

(Don't forget to update the example too)